composer/core/state.py

```diff
@@ -983,11 +986,15 @@ def load_model_state(
             torch.nn.modules.utils.consume_prefix_in_state_dict_if_present(state_dict['model'], 'module.')
 
         try:
-            if self.fsdp_enabled and self.fsdp_state_dict_type is not None:
-                with fsdp_state_dict_type_context(self.model, state_dict_type=self.fsdp_state_dict_type):
+            # Load model if it exists. For FSDP monolith checkpoints, the model does not exist on ranks > 0
+            if 'model' in state_dict:
+                if self.fsdp_enabled and self.fsdp_state_dict_type is not None:
+                    with fsdp_state_dict_type_context(self.model, state_dict_type=self.fsdp_state_dict_type):
+                        missing_keys, unexpected_keys = self.model.load_state_dict(state_dict['model'], strict=strict)
```

Comment:
are we sure this works. I think FSDP would hang because all ranks have to synchronize for a `with fsdp_state_dict_type_context(self.model, state_dict_type=self.fsdp_state_dict_type)` call. I don't think you need it if the model has not been wrapped with fsdp